### PR TITLE
fix: usersWithRole(Owner) returns creators only if privilegedCreatorRole is true

### PR DESCRIPTION
**Pull request by ganfra**
## Content
What the title says.

## Motivation and context
Fix an issue on room version < 12 where creators shouldn't have a privileged creator role.

## Screenshots / GIFs
No ui change.

## Tests

- Open an old room where you have you and the user who created the room as admins.
- See he's displayed only once in the edit admin role.
- 

## Tested devices

- [ ] Physical
- [ ] Emulator
- OS version(s):

## Checklist

- [ ] Changes have been tested on an Android device or Android emulator with API 24
- [ ] UI change has been tested on both light and dark themes
- [ ] Accessibility has been taken into account. See https://github.com/element-hq/element-x-android/blob/develop/CONTRIBUTING.md#accessibility
- [ ] Pull request is based on the develop branch
- [ ] Pull request title will be used in the release note, it clearly define what will change for the user
- [ ] Pull request includes screenshots or videos if containing UI changes
- [ ] You've made a self review of your PR

### CODE REVIEW

File: libraries/matrix/api/src/main/kotlin/io/element/android/libraries/matrix/api/room/RoomInfo.kt [hunk 1]
```diff
@@ -85,7 +85,7 @@ data class RoomInfo(
      */
     fun usersWithRole(role: RoomMember.Role): List<UserId> {
         return if (role is RoomMember.Role.Owner && role.isCreator) {
-            this.creators
+            if (privilegedCreatorRole) this.creators else emptyList()
```
Comment (bmarty) [inline libraries/matrix/api/src/main/kotlin/io/element/android/libraries/matrix/api/room/RoomInfo.kt]: Maybe add a unit test to check that an empty list is returned when the conditions are met?

**Comment by ganfra** [comment]
@bmarty I've actually rework that a bit more : 
I think this is a bit error prone to rely on the RoomInfo and PowerLevel for such things, so now we are only using the member list to compute those info. 
Also this is parity with iOS.

File: libraries/matrix/api/src/main/kotlin/io/element/android/libraries/matrix/api/room/powerlevels/RoomMembersWithRole.kt [hunk 2]
```diff
@@ -34,12 +33,17 @@ fun BaseRoom.usersWithRole(role: RoomMember.Role): Flow<ImmutableList<RoomMember
...
-        .combine(readyMembersFlow) { powerLevels, membersState ->
-            membersState.activeRoomMembers()
-                .filter { powerLevels.contains(it.userId) }
-                .toImmutableList()
-        }
-        .distinctUntilChanged()
+    return readyMembersFlow.map { membersState ->
+        membersState
+            .activeRoomMembers()
+            .filter({ predicate(it.role) })
```
Comment (bmarty) [inline libraries/matrix/api/src/main/kotlin/io/element/android/libraries/matrix/api/room/powerlevels/RoomMembersWithRole.kt]:
```suggestion
            .filter { predicate(it.role) }
```

**Review by bmarty — APPROVED** [review]
Thanks, the code looks a bit clearer.
